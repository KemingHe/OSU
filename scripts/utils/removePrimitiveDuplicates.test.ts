// ./scripts/utils/removePrimitiveDuplicates.test.ts
//
// Unittests for the removePrimitiveDuplicates utility function.

// Vitest essential imports.
import { describe, expect, it } from "vitest";

// Testing module imports.
import removePrimitiveDuplicates from "@scripts/utils/removePrimitiveDuplicates";

// -----------------------------------------------------------------------------
// Test suite for the removePrimitiveDuplicates utility function.
describe("The removePrimitiveDuplicates utility function", () => {
  it("removes duplicate strings from an array", () => {
    const rawStrings: string[] = ["a", "a", "b", "b", "c"];
    const correctStrings: string[] = ["a", "b", "c"];
    const sanitizedStrings: string[] = removePrimitiveDuplicates(rawStrings);

    expect(sanitizedStrings).toStrictEqual(correctStrings);
    expect(sanitizedStrings).toMatchSnapshot();
  });

  it("removes duplicate numbers from an array", () => {
    const rawNumbers: number[] = [1, 1, 2, 2, 3];
    const correctNumbers: number[] = [1, 2, 3];
    const sanitizedNumbers: number[] = removePrimitiveDuplicates(rawNumbers);

    expect(sanitizedNumbers).toStrictEqual(correctNumbers);
    expect(sanitizedNumbers).toMatchSnapshot();
  });
});
