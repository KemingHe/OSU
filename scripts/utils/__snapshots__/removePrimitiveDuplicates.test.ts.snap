// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`The removePrimitiveDuplicates utility function > removes duplicate numbers from an array 1`] = `
[
  1,
  2,
  3,
]
`;

exports[`The removePrimitiveDuplicates utility function > removes duplicate strings from an array 1`] = `
[
  "a",
  "b",
  "c",
]
`;
